library/nsxt_route_advertise.py

```diff
@@ -0,0 +1,181 @@
+    logical_routers = get_logical_routers(module, manager_url, mgr_username, mgr_password, validate_certs)
+    for logical_router in logical_routers['results']:
+        if logical_router.__contains__('display_name') and logical_router['display_name'] == display_name:
+            return logical_router
+    return None
+
+def get_revision(module, manager_url, mgr_username, mgr_password, validate_certs, logical_router_id):
+  try:
+    (rc, resp) = request(manager_url+ '/logical-routers/%s/routing/advertisement' % logical_router_id, headers=dict(Accept='application/json'),
+                      url_username=mgr_username, url_password=mgr_password, validate_certs=validate_certs, ignore_errors=True)
+    return resp['_revision']
+  except Exception as err:
+    module.fail_json(msg='Error accessing current advertisement. Error [%s]' % (to_native(err)))
+    
+
+def main():
+  argument_spec = vmware_argument_spec()
+  argument_spec.update(display_name=dict(required=True, type='str'),
+                        enabled=dict(required=False, type='bool'),
+                        advertise_static_routes=dict(required=False, type='bool'),
+                        advertise_dns_forwarder=dict(required=False, type='bool'),
+                        advertise_lb_snat_ip=dict(required=False, type='bool'),
+                        advertise_lb_vip=dict(required=False, type='bool'),
+                        advertise_nat_routes=dict(required=False, type='bool'),
+                        advertise_nsx_connected_routes=dict(required=False, type='bool')
+                        )
+
+  module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+  advertise_params = get_advertise_params(module.params.copy())
+  mgr_hostname = module.params['hostname']
+  mgr_username = module.params['username']
+  mgr_password = module.params['password']
+  validate_certs = module.params['validate_certs']
+  display_name = module.params['display_name']
+  manager_url = 'https://{}/api/v1'.format(mgr_hostname)
+
+  logical_router_dict = get_lr_from_display_name (module, manager_url, mgr_username, mgr_password, validate_certs, display_name)
+  logical_router_id = None
+  if logical_router_dict:
+    logical_router_id = logical_router_dict['id']
+  
+  advertise_params['_revision'] = get_revision(module, manager_url, mgr_username, mgr_password, 
+                                              validate_certs, logical_router_id) # update current revision
+
+  headers = dict(Accept="application/json")
+  headers['Content-Type'] = 'application/json'
+   
+  # add the pool
+  if module.check_mode:
+    module.exit_json(changed=True, debug_out=str(json.dumps(advertise_params)), id='12345')
+  request_data = json.dumps(advertise_params)
+  try:
+    (rc, resp) = request(manager_url+ '/logical-routers/%s/routing/advertisement' % logical_router_id, data=request_data, headers=headers, method='PUT',
```

Comment:
update only, if there are changes in advertise_params object. Please check other modules, where we update the object only if changes are there.